spack/packages/dla-future/package.py

```diff
@@ -16,23 +15,40 @@ class DlaFuture(CMakePackage):
 
     version('develop', branch='master')
 
-    variant('gpu', default=False,
+    variant('cuda', default=False,
             description='Use the GPU/cuBLAS back end.')
-
-    # Until mpich is default comment this out
+    variant('doc', default=False,
+            description='Build documentation.')
+    
     #depends_on('mpi@3:')
-    depends_on('mpich')
-    depends_on('mkl')
+    depends_on('mpi')
     depends_on('blaspp')
     depends_on('lapackpp')
-    depends_on('hpx cxxstd=14 networking=none')
-    depends_on('cuda', when='gpu=True')
+    depends_on('hpx@1.4.1: max_cpu_count=128 cxxstd=14 networking=none')
```

Comment:
I see that you opened the issue #248 about HPX version. I'm going to look into it. 😉

Reply:
I'm still not able to reproduce the problem in #248, so currently HPX 1.4.0 seems to work correctly with our code.
On the other side, HPX 1.5.0 requires some changes (see #249) so I would add the constraint on the upper version.